src/main/java/org/cyclops/integratedterminals/api/terminalstorage/ITerminalStorageTabClient.java

```diff
@@ -26,6 +26,7 @@
     public static final int DEFAULT_SLOT_OFFSET_Y = 40;
     public static final int DEFAULT_SLOT_VISIBLE_ROWS = 5;
     public static final int DEFAULT_SLOT_ROW_LENGTH = 9;
+    Boolean DEFAULT_AUTO_FOCUS_BEHAVIOUR = false;
```

Comment:
Shall we move this to GeneralConfig to make it configurable?

Reply:
I don't think so but maybe you can shed some more light on this.

I'm recent to the codebase so still learning my way around. I based most of this from how other buttons work and pass data up to the screen.

So [this](https://github.com/nskobelevs/IntegratedTerminals/blob/feature/autofocus/src/main/java/org/cyclops/integratedterminals/client/gui/container/ContainerScreenTerminalStorage.java#L916-L920) function is based on `getSlotRowLength` for example.
If I'm understanding right - we only use the `orEsle` fallback if `getSelectedClientTab` doesn't return a tab which can only happen when there are no interfaces connected to the network. So this isn't the default setting of the button but instead the default behaviour when there are not tabs (and hence no buttons) - do I understand that correctly?

In that case, is there even a purpose to configure that option? There are no tabs and nothing to be searched for so in my opinion turning autofocus off in this case makes the most sense.

We could however add a configuration option for the button itself, which by my understanding would be used [here](https://github.com/nskobelevs/IntegratedTerminals/blob/1c0255d5c0b89d8c3d4fadaf8634db6476651604/src/main/java/org/cyclops/integratedterminals/core/terminalstorage/button/TerminalButtonAutoFocus.java#L49)

Reply:
Ah, I misinterpreted the meaning of that variable. It's not really the default of a button, but the fallback in case no tab data is available.

Perhaps we can rename it to `FALLBACK_AUTO_FOCUS_BEHAVIOUR`?
And I guess it can also be a `public static final boolean`.

Reply:
> We could however add a configuration option for the button itself, which by my understanding would be used [here](https://github.com/nskobelevs/IntegratedTerminals/blob/1c0255d5c0b89d8c3d4fadaf8634db6476651604/src/main/java/org/cyclops/integratedterminals/core/terminalstorage/button/TerminalButtonAutoFocus.java#L49)

Optionally, but not super important I think.